### DISCUSSION TITLE
Using Fabric with Openrouter.ai?

### DISCUSSION THREAD

**Discussion by exponentiallyio** [danielmiessler/Fabric#495] [Q&A] [answered]
Is this possible? It would be amazing to instantly connect up with the whole list of models like Llama, Mistral, Phi:

https://openrouter.ai/models

Thank you

**Comment by Gerkinfeltser** [comment, marked as answer]
Hi, I've been using openrouter successfully via this method (if I'm remembering my process correctly). All that was necessary was to input the relevant information into the `~/.config/fabric/.env` file.

### A few notes:
- The `OPENAI_API_KEY` takes an openrouter key you generate on the [openrouter account keys page](https://openrouter.ai/keys).
- Correct model names can be grabbed from the openrouter model pages. For instance, on the [llama-3-70b-instruct](https://openrouter.ai/models/meta-llama/llama-3-70b-instruct) page, you can get the name by clicking the little clipboard icon under the model title (which gives you `meta-llama/llama-3-70b-instruct`). This value can be used as I have it here in the .env as `DEFAULT_MODEL`  or you can pass it to fabric via the `-m` arg (`fabric -m meta-llama/llama-3-70b-instruct -p summarize`).
- fwiw I've been getting satisfactory responses from `meta-llama/llama-3-8b-instruct:free`, which is good cause I'm cheap. 😉 

### Example ~/.config/fabric/.env
```~/.config/fabric/.env
DEFAULT_MODEL=meta-llama/llama-3-8b-instruct:free
OPENAI_BASE_URL=https://openrouter.ai/api/v1
OPENAI_API_KEY=sk-or-v1-yourOpenrouterKeyHere
```

**Reply by lukaDen** [reply]
Could you use smart router option as default model maybe?

**Comment by exponentiallyio** [comment]
Thank you! That worked great

**Comment by quantdev** [comment]
After the rewrite in golang, I do not think this works anymore.

**Comment by rudi-bruchez** [comment]
Hello, I added OpenRouter as a new vendor, because I use it too ;)
And created a PR
